### PR TITLE
JAMES-3429 Add StateMismatch error type and getLatestVersion FilteringManagement

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

**Comment by chibenwa** [comment]
Drop 124644c commit

File: server/data/data-jmap/src/main/java/org/apache/james/jmap/api/exception/StateMismatchException.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ ****************************************************************/
+
+package org.apache.james.jmap.api.exception;
+
+public class StateMismatchException extends RuntimeException {
+    public StateMismatchException(String msg) {
+        super(msg);
+    }
+
+    public static void checkState(boolean expression, String msg) {
```
Comment (chibenwa) [inline server/data/data-jmap/src/main/java/org/apache/james/jmap/api/exception/StateMismatchException.java:27]: static methods come first

File: server/data/data-jmap/src/main/java/org/apache/james/jmap/api/exception/StateMismatchException.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.api.exception;
+
+public class StateMismatchException extends RuntimeException {
```
Comment (chibenwa) [inline server/data/data-jmap/src/main/java/org/apache/james/jmap/api/exception/StateMismatchException.java:22]: Maybe we can extend IllegalStateException ?

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Review by LanKhuat — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by quantranhong1999** [comment]
Tests faild on CI:
```
org.apache.james.jmap.cassandra.filtering.CassandraEventSourcingFilteringManagementTest.setRulesWithIfInStateIsOneWhenNonStateIsDefinedShouldFail{EventStore}
org.apache.james.jmap.cassandra.filtering.CassandraEventSourcingFilteringManagementTest.setRulesWithIfInStateIsInitialWhenAStateIsDefinedShouldFail{EventStore}
org.apache.james.jmap.cassandra.filtering.CassandraEventSourcingFilteringManagementTest.modifyExistingRulesWithWrongCurrentVersionShouldFail{EventStore}
```
But green on local, please rerun the build

**Comment by Arsnael** [comment]
Build restarted

**Comment by quantranhong1999** [comment]
Don't know why this keep failing, just a rebase and make the ci run again.

**Pull request merged**